### PR TITLE
#1445: Refactor admin controller tests. Remove all before_all blocks,…

### PR DESCRIPTION
**Pull request by murny**
… and other refactors

#1445 

First of many PRs. This one just tackles our admin/controller tests.

A quick example of how minor this change is amd helps illustrate how quick setup and fixtures are:

Using `before_all` to create objects before all tests which are then used for each test.

```
➜  jupiter (integration_postmigration) ✗ rails test test/controllers/admin/theses/draft_controller_test.rb
Run options: --seed 8096

# Running:

..............

Finished in 3.027713s, 4.6240 runs/s, 15.8536 assertions/s.
14 runs, 48 assertions, 0 failures, 0 errors, 0 skips
Coverage report generated for Unit Tests to /Code/jupiter/coverage. 1404 / 2670 LOC (52.58%) covered.
```

Using `setup` and fixtures, to tear down the DB and reprovision the DB with all the fixture data before each and every test:
```
➜  jupiter (integration_postmigration) ✗ rails test test/controllers/admin/theses/draft_controller_test.rb
Run options: --seed 53600
# Running:

..............

Finished in 3.037624s, 4.6089 runs/s, 15.8018 assertions/s.
14 runs, 48 assertions, 0 failures, 0 errors, 0 skips
Coverage report generated for Unit Tests to /Code/jupiter/coverage. 1398 / 2670 LOC (52.36%) covered.
```

Difference is hardly even noticeable

NOTE: Travis is hating on this PR. Looking into this. Seems to have been fixed...maybe caching issue? will have to keep an eye out for this.

### CODE REVIEW

File: test/controllers/admin/communities_controller_test.rb [hunk 1]
```diff
@@ -29,6 +23,12 @@ def setup
     assert_response :success
   end
 
+  test 'should show community for js response' do
```
Comment (murny) [inline test/controllers/admin/communities_controller_test.rb]: Saw we were missing a test for this. Added one.

File: test/controllers/admin/theses/files_controller_test.rb [hunk 2]
```diff
@@ -30,10 +24,9 @@ def before_all
   end
 
   test 'should be able to attach files to a draft thesis' do
-    # TODO: Why this failing?
```
Comment (murny) [inline test/controllers/admin/theses/files_controller_test.rb]: Seems to be working now? Maybe now that we are off monkey patched ActiveStorage or something?

File: test/fixtures/collections.yml [hunk 3]
```diff
@@ -2,12 +2,18 @@ fantasy_books:
   title: 'Fantasy Books'
   owner: admin
   community: books
+  record_created_at: <%= Time.zone.now - 1.hour %>
```
Comment (murny) [inline test/fixtures/collections.yml]: These are required fields for Community/Collections. Added them to fixtures

File: test/controllers/admin/items_controller_test.rb [hunk 4]
```diff
@@ -46,8 +14,6 @@ def setup
   end
 
   test 'should destroy item and its derivatives' do
-    # TODO: 'ActiveStorage::Attachment.count' || 'ActiveStorage::Blob.count'
-    # wish I could test the thumbnail deletion but flaky on travis-ci
```
Comment (pgwillia) [inline test/controllers/admin/items_controller_test.rb]: I think we should leave this in because the TODO hasn't been addressed yet.

Comment (murny) [reply test/controllers/admin/items_controller_test.rb]: Yeah I could go either way. I choose the simple solution. We don't need attachments for these test as we are basically just doing simple smoke test to see if, we can load the index and that we can delete an item successfully. 

If deleting an item cleanups its attachments is important (which i say it is) than that should probably be well tested in the model. 

But its not much more effort to double check that here either. So I could add that back. I noticed I never updated the test name `should destroy item and its derivatives` so regardless of what I choose, I should do that 😆

Comment (murny) [reply test/controllers/admin/items_controller_test.rb]: This is fixed now. Works locally, will keep an eye on it for TravisCI in case it starts flaking

**Comment by murny** [comment]
Too many flakey tests, not able to get this passing. #1558 fixes these flakey tests completely. So closing this for https://github.com/ualbertalib/jupiter/pull/1558 instead. Moving additional fixes I made over to that PR

**Pull request closed without merging**